src/FluentCheck.ts

```diff
@@ -8,7 +8,8 @@ type WrapFluentPick<T> = { [P in keyof T]: FluentPick<T[P]> }
 type FluentPicks = Record<string, FluentPick<any> | any>
 
 export class FluentResult {
-  constructor(public readonly satisfiable = false, public example: FluentPicks = {}) { }
+  constructor(public readonly satisfiable = false, public readonly example: FluentPicks = {},
+    public readonly testCases: Set<String> = new Set()) { }
```

Comment:
I don't think you should use a `Set`, otherwise you risk deduplicating _duplicated_ test cases. In fact, I think "(%) of duplicated tests" would be a great metric to report later.

Reply:
I agree, it can be an interesting metric